src/main/kotlin/no/fdk/dataset_catalog/service/DatasetService.kt

```diff
@@ -69,29 +68,48 @@ class DatasetService(
                 registrationStatus = dataset.registrationStatus ?: REGISTRATION_STATUS.DRAFT)
             .updateConcepts()
             .updateSubjects()
+            .allAffectedSeriesDatasets(null)
             .let { persistAndHarvest(it, catalog) }
+
+        return getByID(catalogId, datasetId)
     }
 
     fun count() = datasetRepository.count()
 
     fun updateDataset(catalogId: String, id: String, operations: List<JsonPatchOperation>): Dataset? {
         val dataset = getByID(catalogId, id)
 
-        return dataset
-            ?.update(operations)
+        dataset?.update(operations)
             ?.copy(
                 id = id,
                 catalogId = catalogId,
                 specializedType = dataset.specializedType,
                 lastModified = LocalDateTime.now())
             ?.updateConcepts()
             ?.updateSubjects()
+            ?.allAffectedSeriesDatasets(dataset)
             ?.let { persistAndHarvest(it, catalogService.getByID(catalogId)) }
+
+        return getByID(catalogId, id)
     }
 
     fun delete(catalogId: String, id: String) {
         getByID(catalogId, id)
-            ?.let { datasetRepository.delete(it) }?: throw Exception()
+            ?.also { datasetRepository.delete(it) }
+            ?.removeDeletedDatasetFromSeriesFields()
+            ?: throw Exception()
+    }
+
+    private fun Dataset.removeDeletedDatasetFromSeriesFields() {
+        if (id != null) {
+            inSeries?.let { datasetRepository.findAllById(it) }
+                    ?.map { it.copy(seriesOrder = it.seriesOrder?.minus(id)) }
```

Comment:
Skulle man også endre sort verdien med -1?

Reply:
Er det sånn supernødvendig? hvis hull så kan vi jo bare hoppe videre til neste. Vil uansett poppe opp lignende problem om det har blitt lagt til et datasett som ikke er publisert

Reply:
Jeg tenkte i hvert fall at det ver greit å ikke være så supernøyaktig på at int-verdien 100% gjenspeilet indeksen i den ferdige serien, bare at vi brukte den som en vekting for sorteringsrekkefølgen. Begynte å bli en del ekstra som måtte håndteres om int-verdien, spesielt mtp eventuelle etl-jobber, alltid skulle være korrekt serie-indeks...